flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/sink/GlobalStreamingCommitterOperatorTest.java

```diff
@@ -121,13 +119,18 @@ public void restoredFromMergedState() throws Exception {
 		testHarness.open();
 
 		final List<String> expectedOutput = new ArrayList<>();
-		expectedOutput.add(TestSink.TestGlobalCommitter.COMBINER.apply(input1));
-		expectedOutput.add(TestSink.TestGlobalCommitter.COMBINER.apply(input2));
+		expectedOutput.add(TestSink.DefaultGlobalCommitter.COMBINER.apply(input1));
+		expectedOutput.add(TestSink.DefaultGlobalCommitter.COMBINER.apply(input2));
 
 		testHarness.snapshot(1L, 1L);
 		testHarness.notifyOfCompletedCheckpoint(1L);
 		testHarness.close();
 
+		// TODO:: maybe there is no output at all
```

Comment:
What's happening here?

Reply:
This should be removed. The base committer class uses the `output` although it is only needed by the `Committer` and not the `GlobalCommitter`. This does not seem to be a bug because every operator has an output but if there is no successor, the elements are simply dropped.

Reply:
thanks @aljoscha and @kl0u . It would be fix in the next pr.